Prox/Prox/Utilities/CategoriesUtil.swift

```diff
@@ -124,6 +124,33 @@ struct CategoriesUtil {
         }
         return categoryToName
     }()
+
+    static let categoryToFilter: [String: Filter] = {
+        var categoryToFilter = [String: Filter]()
+
+        // Build the initial map of categories -> filters that we set explicitly.
+        for (filter, categories) in Filter.categories {
+            for category in categories {
+                categoryToFilter[category] = filter
+            }
+        }
+
+        // Set the remaining categories based on the first ancestor with an explicitly-set filter.
+        for category in categoryToParentsMap.keys {
+            guard let firstCategorizedParent = getAncestors(forCategory: category).flatMap({ categoryToFilter[$0] }).first else { continue }
+            categoryToFilter[category] = firstCategorizedParent
+        }
+
+        return categoryToFilter
+    }()
+
+    /// Returns this category, its root categories, and all parents in between.
+    /// The result is DFS-ordered.
```

Comment:
Anthony and I were looking through the categories yesterday and found that the first category often seems to best represent a place. Perhaps that was luck? I hope not since we now associate places with a single filter, and we wouldn't have a good way to choose which parent to use.